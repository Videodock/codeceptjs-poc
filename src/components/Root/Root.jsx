import React from 'react';
import { Route, Switch } from 'react-router-dom';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import useTheme from '@material-ui/core/styles/useTheme';
import Home from '../../screens/Home';
import Task from '../../screens/Task';
import ActiveCampaign from '../../screens/ActiveCampaign';
import NotFound from '../../screens/NotFound';
import WhoAreYou from '../../screens/WhoAreYou';
import NoSupportMobile from '../../screens/NoSupportMobile';

export default function Root(props) {
  const theme    = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.only('xs'));

  if(isMobile) {
    return <NoSupportMobile />;
  }

  const content = props.error ? (
    <div>Something wen't terribly wrong!</div>
  ) : (
    <Switch>
      <Route path="/" component={Home} exact />
      <Route path="/campaign/:campaignIdentifier" component={ActiveCampaign} exact />
      <Route path="/campaign/:campaignIdentifier/who-are-you" component={WhoAreYou} exact />
      <Route path="/campaign/:campaignIdentifier/task/" component={Task} exact />
      <Route path="/campaign/:campaignIdentifier/task/:taskIdentifier" component={Task} exact />
      <Route component={NotFound} />
    </Switch>
  );

  return (
    <div>
      {content}
    </div>
  );
};
