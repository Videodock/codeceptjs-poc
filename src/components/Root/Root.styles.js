import { createStyles } from '@material-ui/styles';

export default ({ breakpoints }) => createStyles({
  container: {
    [breakpoints.up('xl')]: {
      width : 1860,
      margin: [0, 'auto'],
    },
    width: '100%',
  },
});
