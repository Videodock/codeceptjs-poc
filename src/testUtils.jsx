import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <I18nextProvider i18n={i18n}>
        <Router>
          {children}
        </Router>
      </I18nextProvider>
    </ThemeProvider>
  </Provider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
