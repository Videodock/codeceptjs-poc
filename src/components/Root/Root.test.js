import React from 'react';
import { render } from '../../testUtils';
import Root from './Root';

describe('<Root />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Root />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders a generic error page when error prop is given', () => {
    const { container } = render(<Root error />);

    expect(container.firstChild).toHaveTextContent('Something wen\'t terribly wrong!');
    expect(container.firstChild).toMatchSnapshot();
  });
});
