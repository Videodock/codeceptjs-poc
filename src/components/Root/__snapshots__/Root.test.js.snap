// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> renders a generic error page when error prop is given 1`] = `
<div
  class="makeStyles-container-34"
>
  <div>
    Something wen't terribly wrong!
  </div>
</div>
`;

exports[`<Root /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-container-1"
>
  <div>
    <div
      class="makeStyles-hero-3"
    >
      <div>
        <img
          alt="logo"
          class="makeStyles-logo-2"
          src="[object Object]"
          width="400"
        />
      </div>
      <div>
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-colorInherit"
        >
          Welcome to the Videodock React Redux Boilerplate website!
        </h4>
      </div>
      <div>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
        >
          this is awwwwwee.... wait for it... some!
        </p>
      </div>
    </div>
  </div>
</div>
`;
